perceval/backends/core/pagure.py

```diff
@@ -0,0 +1,404 @@
+        """
+        return True
+
+    @classmethod
+    def has_resuming(cls):
+        """Returns whether it supports to resume the fetch process.
+
+        :returns: this backend supports items resuming
+        """
+        return True
+
+    @staticmethod
+    def metadata_id(item):
+        """Extracts the identifier from a Pagure item."""
+
+        return str(item['id'])
+
+    @staticmethod
+    def metadata_updated_on(item):
+        """Extracts the update time from a Pagure item.
+
+        The timestamp used is extracted from 'last_updated' field.
+        This date is converted to UNIX timestamp format. As Pagure
+        dates are in timestamp format the conversion is straightforward.
+
+        :param item: item generated by the backend
+
+        :returns: a UNIX timestamp
+        """
+        ts = int(item['last_updated'])
+        ts = datetime.fromtimestamp(ts).timestamp()
+
+        return ts
+
+    @staticmethod
+    def metadata_category(item):
+        """Extracts the category from a Pagure item.
+
+        This backend generates one type of item which is
+        'issue'.
+        """
+        category = CATEGORY_ISSUE
+
+        return category
+
+    def _init_client(self, from_archive=False):
+        """Init client"""
+
+        return PagureClient(self.namespace, self.repository, self.api_token,
+                            self.sleep_time, self.max_retries, self.max_items,
+                            self.archive, from_archive, self.ssl_verify)
+
+    def __fetch_issues(self, from_date, to_date):
+        """Fetch the issues
+        :param from_date: starting date from which issues are fetched
+        :param to_date: ending date till which issues are fetched
+
+        :returns: an issue object
+        """
+        issues_groups = self.client.issues(from_date=from_date)
+
+        for raw_issues in issues_groups:
+            issues = json.loads(raw_issues)
+            issues = issues['issues']
+            for issue in issues:
+
+                if int(issue['last_updated']) > str_to_datetime(to_date).timestamp():
+                    return
+
+                yield issue
+
+
+class PagureClient(HttpClient, RateLimitHandler):
+    """Client for retieving information from Pagure API
+
+    :param namespace: Pagure namespace
+    :param repository: Pagure repository; incase the repository is within a namespace
+    :param token: Pagure API token to access the API
+    :param sleep_time: time to sleep in case
+        of connection problems
+    :param max_retries: number of max retries to a data source
+        before raising a RetryError exception
+    :param max_items: max number of category items (e.g., issues,
+        pull requests) per query
+    :param archive: collect issues already retrieved from an archive
+    :param from_archive: it tells whether to write/read the archive
+    :param ssl_verify: enable/disable SSL verification
+    """
+    EXTRA_STATUS_FORCELIST = [403, 500, 502, 503]
```

Comment:
This can be removed. ATM we don't know which extra status should be defined (no info in the API).